### ISSUE TITLE
Update-PSResource fails with required nuget version

### ISSUE THREAD

**Issue by ThomasNieto** [PowerShell/PSResourceGet#1311]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSResourceGet)
- [X] Search the existing issues.

### Steps to reproduce

If you try and `Update-PSResource` using a nuget version syntax it fails, `Install-PSResource` works. This workflow worked in beta22 so something has changed to break it.

I found this while running CI for my module: [CI run](https://github.com/anypackage/psresourceget/actions/runs/5682985708/job/15402602951)

```powershell
PS C:\Users\Thomas> Install-PSResource -Name NetworkingDsc -Version '[8.0.0]' -PassThru

Name          Version Prerelease Repository Description
----          ------- ---------- ---------- -----------
NetworkingDsc 8.0.0              PSGallery  DSC resources for configuring settings related to networking.

PS C:\Users\Thomas> Update-PSResource -Name NetworkingDsc -Version '[8.1.0]' -PassThru

Name          Version Prerelease Repository Description
----          ------- ---------- ---------- -----------
NetworkingDsc 8.1.0              PSGallery  DSC resources for configuring settings related to networking.

PS C:\Users\Thomas> Update-PSResource -Name NetworkingDsc -Version '[8.2.0-preview0001]' -PassThru
Update-PSResource: Package 'NetworkingDsc' with version '[8.2.0-preview0001]' could not be found
WARNING: Installed package 'NetworkingDsc':'[8.2.0-preview0001]' was not found in repositories and cannot be updated.

PS C:\Users\Thomas> Update-PSResource -Name NetworkingDsc -Version '[8.2.0-preview0001]' -PassThru -Prerelease
Update-PSResource: Package 'NetworkingDsc' with version '[8.2.0-preview0001]' could not be found
WARNING: Installed package 'NetworkingDsc':'[8.2.0-preview0001]' was not found in repositories and cannot be updated.
```

### Expected behavior

```console
# works
```

### Actual behavior

```console
Update-PSResource -Name NetworkingDsc -Version '[8.2.0-preview0001]' -PassThru -Prerelease
Update-PSResource: Package 'NetworkingDsc' with version '[8.2.0-preview0001]' could not be found
WARNING: Installed package 'NetworkingDsc':'[8.2.0-preview0001]' was not found in repositories and cannot be updated.
```

### Error details

```console
Exception             :
    Type           : Microsoft.PowerShell.PSResourceGet.UtilClasses.PackageNotFoundException
    Message        : Package 'NetworkingDsc' with version '[8.2.0-preview0001]' could not be found
    InnerException :
        Type    : Microsoft.PowerShell.PSResourceGet.UtilClasses.V2ResourceNotFoundException
        Message : Package does not exist on the server
        HResult : -2146233088
    HResult        : -2146233088
TargetObject          : Microsoft.PowerShell.PSResourceGet.Cmdlets.FindHelper
CategoryInfo          : InvalidResult: (Microsoft.PowerShel….Cmdlets.FindHelper:FindHelper) [Update-PSResource],
PackageNotFoundException
FullyQualifiedErrorId : FindNameConvertToPSResourceFailure,Microsoft.PowerShell.PSResourceGet.Cmdlets.UpdatePSResource
InvocationInfo        :
    MyCommand        : Update-PSResource
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : 4
    Line             : Update-PSResource -Name NetworkingDsc -Version '[8.2.0-preview0001]' -PassThru -Prerelease
    Statement        : Update-PSResource -Name NetworkingDsc -Version '[8.2.0-preview0001]' -PassThru -Prerelease
    PositionMessage  : At line:1 char:1
                       + Update-PSResource -Name NetworkingDsc -Version '[8.2.0-preview0001]'  …
                       + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : Update-PSResource
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo :
```

### Environment data

```powershell
beta23
```

**Comment by SydneyhSmith** [comment]
Thanks @ThomasNieto, we should also check our tests, as a test should have caught this regression

**Comment by ThomasNieto** [comment]
@SydneyhSmith feel free to compare to my tests for [anypackage/psresourceget](https://github.com/anypackage/psresourceget/tree/main/test) to see if there are any gaps.

**Comment by alerickson** [comment]
Just working on the bug fix for this we did have a [test](https://github.com/PowerShell/PSResourceGet/blob/5eaa49942b1d3634e84ad341a5a61ad9b1cb08c2/test/UpdatePSResourceTests/UpdatePSResourceV2Tests.ps1#L157) to check for this scenario.  Running the test locally it fails before the bug fix and is resolved after the bug fix.  I don't know how this test didn't fail in CI, but from what I'm looking at it both should always be a test that's run and should definitely have failed.

**Comment by ThomasNieto** [comment]
In that linked test I don't see where it is specifying the required version using the square bracket syntax. Am I missing something?

**Comment by alerickson** [comment]
@ThomasNieto to backtrack a bit, I was looking at tests to see if we had anything to catch this bug and I saw the test I linked to above.  It's different than your original issue but fails because of the same underlying problem (the `-Prerelease` param not getting passed through).  So that's the context around that comment.  Ideally this test should have caught that regression but for some reason didn't (ie it's failing locally as it should and passing in CI even though it shouldn't).  

The reason why the test linked to is similar to your issue is because they both treat the version for the package to be updated to as a version range, which follows a different code path than a "specific version" (ie "9.0.0" instead of "[9.0.0]"). 

I opened a dummy PR [here](https://github.com/PowerShell/PSResourceGet/pull/1321), to try to figure out why CI is giving a false positive, in the hopes of properly fixing that test and having it act as a (properly working) test for the linked PR I opened, but I still can't quite figure out what's going on there.  So I'll just add a test to the PR with the bracket syntax.
